controllers/meshgateway_controller.go

```diff
@@ -39,15 +55,161 @@ type MeshGatewayReconciler struct {
 
 func (r *MeshGatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = context.Background()
-	_ = r.Log.WithValues("meshgateway", req.NamespacedName)
+	logger := r.Log.WithValues("meshgateway", req.NamespacedName)
+
+	mgw := &servicemeshv1alpha1.MeshGateway{}
+	err := r.Get(context.TODO(), req.NamespacedName, mgw)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("reconciling")
+
+	reconciler, err := GetHelmReconciler(r, meshgateway.NewChartReconciler)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	icp, err := r.getRelatedIstioControlPlane(ctx, r.GetClient(), mgw, logger)
+	// do not reqeue if control plane was not found
+	if icp == nil && err == nil {
+		return ctrl.Result{
+			Requeue: false,
+		}, nil
+	}
+
+	if !isIstioVersionSupported(icp.Spec.Version) {
+		return ctrl.Result{}, nil
+	}
```

Comment:
Shouldn't we put `requeue: false` on the result and also log an error maybe?
this is from icp reconciler:
```suggestion
	if !isIstioVersionSupported(icp.Spec.Version) {
		err = errors.New("intended Istio version is unsupported by this version of the operator")
		logger.Error(err, "", "version", icp.Spec.Version)

		return reconcile.Result{
			Requeue: false,
		}, nil
	}
```

Reply:
the requeue is false by default. i rather return blindly otherwise there will be a lot of unnecessary noise in the log for every unsupported mgw.

Reply:
maybe we can add a debug level log here.

Reply:
I suggested the `requeue: false` bc of this comment here: https://github.com/banzaicloud/istio-operator/pull/713/files/1f902a16213113c8afa64ca641b5287ef2fe7b52#diff-84759ff645b6f0144831537e6dba955fce69a791552223c0626647eb45b9bb65R80-R84 I guess it's not needed there either then, but nvm.

I'd argue with the error log though. There shouldn't be a lot of unsupported MGWs or if there is then the user should be notified about why it is unsupported and probably they'll want to fix the version. If it silently fails or only can be found in debug logs (which are hidden by default), then it'd be hard to find the issue. We logged these kind of errors in the first version of the operator as well and I think it was useful.